core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -848,6 +848,8 @@ class ReplicaManager(val config: KafkaConfig,
                    Errors.COORDINATOR_NOT_AVAILABLE |
                    Errors.NOT_COORDINATOR => Some(new NotEnoughReplicasException(
                 s"Unable to verify the partition has been added to the transaction. Underlying error: ${error.toString}"))
+              case Errors.UNKNOWN_PRODUCER_ID => Some(new OutOfOrderSequenceException(
```

Comment:
Can we add a comment why we translate this error this way?  Also, it looks like we don't return this error any more (at least not in the product code).